docs/reference.md

```diff
@@ -44,4 +44,70 @@ Options:
   run with set -x
 ```
 
-## 
+# Generating Disk Images with Secrets Included in Disko
+
+Using Disko on NixOS allows you to efficiently create `.raw` VM images from a
+system configuration. The generated image can be used as a VM or directly
+written to a physical drive to create a bootable disk. Follow the steps below to
+generate disk images:
+
+## Generating the `.raw` VM Image
+
+1. **Modify the script name:** Replace `mySystem` in the command below with your
+   specific system configuration name:
+   ```bash
+   nix build .#nixosConfigurations.mySystem.config.system.build.diskoImagesScript
+   ```
+2. **Execute the result file:** Execute the generated result file. Running
+   `./result --help` will output the available options:
+
+   ```
+   ./result --help
+   Usage: $script [options]
+
+   Options:
+   * --pre-format-files <src> <dst>
+     copies the src to the dst on the VM, before disko is run
+     This is useful to provide secrets like LUKS keys, or other files you need for formating
+   * --post-format-files <src> <dst>
+     copies the src to the dst on the finished image
+     These end up in the images later and is useful if you want to add some extra stateful files
+     They will have the same permissions but will be owned by root:root
+   * --build-memory
+     specify the ammount of memory that gets allocated to the build vm (in mb)
+     This can be usefull if you want to build images with a more involed NixOS config
+     By default the vm will get 1024M/1GB
+   * --write-to-disk </dev/disk>
+     use an actuall disk instead of writing to a file
+     This only works if your conifg has only one disk specified
+     There is no check if the specified path is actually a disk so you can also write to another file
+   ```
+
+## Additional Configuration
+
+- For virtual drive use, define the image size in your Disko configuration:
```

Comment:
```suggestion
- For virtual drives, define the image size in your Disko configuration:
```

Reply:
technically the other one is also correct, but maybe more confusing.